### PR TITLE
refactor: target s3 prefix calculation in adwords reports to s3 util

### PR DESCRIPTION
**Pull request by vbmudalige**
### What changes are proposed in this PR?
---
* Added get_target_prefix_with_partition_dirs function with unit tests

### How was this tested?
---
* Unit tests

### CODE REVIEW

File: hip_data_tools/etl/adwords_to_athena.py [hunk 1]
```diff
@@ -34,9 +34,9 @@ class AdWordsToAthena(AdWordsToS3):
     def __init__(self, settings: AdWordsToAthenaSettings):
         self.__settings = settings
         self.base_dir = settings.target_key_prefix
-        if self.__settings.is_partitioned_table:
-            partition_dirs = "/".join([f"{k}={v}" for k, v in settings.partition_values])
-            settings.target_key_prefix = f"{settings.target_key_prefix}/{partition_dirs}"
+        settings.target_key_prefix = get_target_prefix_with_partition_dirs(
+            target_key_prefix=settings.target_key_prefix,
+            partition_values=settings.partition_values)
         super().__init__(settings)
```
Comment (arihantsurana) [inline hip_data_tools/etl/adwords_to_athena.py:40]:
```suggestion
        settings.target_key_prefix = get_target_prefix_with_partition_dirs()
        super().__init__(settings)
```

File: hip_data_tools/etl/adwords_to_athena.py [hunk 2]
```diff
@@ -166,3 +165,17 @@ def _construct_athena_table_settings(self, data: DataFrame) -> dict:
...
+    """
+    Calculate the target s3 key prefix including partition directories
+    :param target_key_prefix:
+    :param partition_values:
+    :return:
+    """
+    if partition_values:
+        partition_dirs = "/".join([f"{k}={v}" for k, v in partition_values])
+        return f"{target_key_prefix}/{partition_dirs}"
+    return target_key_prefix
```
Comment (arihantsurana) [inline hip_data_tools/etl/adwords_to_athena.py:181]:
```suggestion
def get_target_prefix_with_partition_dirs() -> str:
    """
    Calculate the target s3 key prefix including partition directories
    fix docstring, use the google python standard
    """
    if self.__settings.partition_values:
        partition_dirs = "/".join([f"{k}={v}" for k, v in self.__settings.partition_values])
        return f"{target_key_prefix}/{partition_dirs}"
    return target_key_prefix
```

**Review by arihantsurana — CHANGES_REQUESTED** [review]

File: hip_data_tools/etl/adwords_to_athena.py (hunk 1 above)
Comment (vbmudalige) [reply hip_data_tools/etl/adwords_to_athena.py]: fixed

File: hip_data_tools/etl/adwords_to_athena.py (hunk 2 above)
Comment (vbmudalige) [reply hip_data_tools/etl/adwords_to_athena.py]: fixed

**Review by arihantsurana — APPROVED** [review]

**Comment by willmeet** [comment]
[How to get $350 threshold from google adwords without credit card or bank account Free Trick 2020](http://www.technologymanias.com/2020/10/how-to-get-350-threshold-from-google.html)

**Pull request merged**